modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionsExchangeFuture.java

```diff
@@ -2402,6 +2407,61 @@ private String exchangeTimingsLogMessage(String header, List<String> timings) {
         return false;
     }
 
+    /**
+     * Update maximum amount of nodes available for safe stop metric.
+     */
+    private void updateMaximumAmountOfNodesAvailableForSafeStop() {
+        int nodeSpecificAmountOfOwners = -1;
+        for (CacheGroupContext grpCtx : cctx.cache().cacheGroups()) {
+            if (grpCtx.systemCache())
+                continue;
+
+            if (grpCtx.isLocal())
+                continue;
+
+            if (grpCtx.config().getCacheMode() == PARTITIONED && grpCtx.config().getBackups() == 0)
+                continue;
+
+            GridDhtPartitionFullMap fullMap = grpCtx.topology().partitionMap(false);
+
+            int cacheSpecificAmountOfOwners = -1;
+
+            if (fullMap != null) {
+                int parts = grpCtx.topology().partitions();
+
+                for (int part = 0; part < parts; part++) {
+                    Set<UUID> partIdealAssignment = grpCtx.affinity().idealAssignmentRaw().get(part).
+                        stream().map(ClusterNode::id).collect(Collectors.toSet());
+
+                    int cnt = 0;
+
+                    for (Map.Entry<UUID, GridDhtPartitionMap> entry : fullMap.entrySet()) {
+                        // Skip all non-ideal assignments.
+                        if (!partIdealAssignment.contains(entry.getKey()))
+                            continue;
+
+                        if (entry.getValue().get(part) == GridDhtPartitionState.OWNING)
+                            cnt++;
+                    }
+
+                    if (cacheSpecificAmountOfOwners == -1)
+                        cacheSpecificAmountOfOwners = cnt;
+
+                    cacheSpecificAmountOfOwners = Math.min(cacheSpecificAmountOfOwners, cnt);
+                }
+            }
+
+            if (nodeSpecificAmountOfOwners == -1)
+                nodeSpecificAmountOfOwners = cacheSpecificAmountOfOwners;
```

Comment:
What's the point for having two variables for storing intermediate minimum?

Reply:
There's no sense in having two variables. Fixed.